File: 프로그래머스/3/42627. 디스크 컨트롤러/디스크 컨트롤러.js
function solution(jobs) {
    let answer = 0;
    let n = jobs.length;
    let time = 0;
    
    jobs.sort((a, b) => a[0] - b[0]);
    
    while (jobs.length > 0) {
        const waiting = jobs.filter(job => job[0] <= time);

        if (waiting.length === 0) time = jobs[0][0];
        else {
            waiting.sort((a, b) => a[1] - b[1]);
            
            const [req, usage] = waiting[0];
            time += usage;
            answer += time - req;
            
            jobs = jobs.filter(job => job[0] !== req || job[1] !== usage);
        }
    }
    
    return (answer / n) >> 0;
}

File: 프로그래머스/3/42627. 디스크 컨트롤러/디스크 컨트롤러.py
def solution(jobs):
    n = len(jobs)
    answer = 0
    time = 0
    
    jobs.sort(key=lambda x: x[0])
    
    while len(jobs) > 0:
        waiting = [job for job in jobs if job[0] <= time]
        
        if len(waiting) == 0:
            time = jobs[0][0]
        else:
            waiting.sort(key=lambda x: x[1])
            
            req, usage = waiting[0]
            time += usage
            answer += time - req
            
            jobs = [job for job in jobs if job[0] != req or job[1] != usage]
    
    return answer // n
